cypress/pageObjects/changeOfNamePage.js

```diff
@@ -101,32 +139,120 @@ class ChangeOfNamePageObjects {
+    }
+
+    missedAppointmentLink() {
+        return cy.get('.lbh-link').contains('Office appointment missed - reschedule');
+    }
+
+    missedAppointmentClose() {
+        return cy.get('.lbh-link').contains('Office appointment missed - close case');
+    }
+
     ahmDecisionApprove(){
         return cy.get('#ho-review-approve');
     };
     ahmDecisionDecline(){
         return cy.get('#ho-review-decline');
     };
-    ahmConfirmBox() {
+
+    checkBoxAhmApproved() {
         return cy.get('#confirm-form-group-field');
     };
+
     ahmNameInput() {
         return cy.get('#area-housing-manager-name-form-group-field');
     };
+
+    ahmConfirmButton() {
+        return cy.get('.lbh-button').contains('Confirm');
+    };
+
     buttonSubmitCase(){
         return cy.get('.mtfh-layout__main > :nth-child(6)');
     };
     errorAreaHousingManagerName() {
         return cy.get('#area-housing-manager-name-form-group-error')
     };
-    buttonReturnToApplication() {
+    modalReturn() {
         return cy.get('[data-testid=close-update-contact-details-modal]');
     };
     emailAddress() {
         return cy.get('#contact-details-email-address-field');
     }
+
+    phoneNumberType() {
+        return cy.get('#contact-details-phone-type-0-field');
+    }
+
+    phoneNumber() {
+        return cy.get('#contact-details-phone-number-0-field');
+    }
+
+    setAppointmentDateTime(date) {    
+        // Set the date
+        this.appointmentDay().clear().type(date.getDate());
+        this.appointmentMonth().clear().type(date.getMonth() + 1); // Month is 0-indexed
+        this.appointmentYear().clear().type(date.getFullYear());
+    
+        // Convert the time to 12-hour format
+        const appointmentHour = date.getHours();
+        const appointmentMinute = date.getMinutes();
+        const appointmentAmPm = appointmentHour >= 12 ? 'PM' : 'AM';
+        const appointmentHour12 = appointmentHour % 12 || 12;
+    
+        // Set the time
+        this.appointmentHour().clear().type(appointmentHour12);
+        this.appointmentMinute().clear().type(appointmentMinute);
+        this.appointmentAmPmSelector().select(appointmentAmPm);
+    }
+
+    appointmentDay() {
+        return cy.get('[name="day"]')
+    }
+
+    appointmentMonth() {
+        return cy.get('[name="month"]')
+    }
+
+    appointmentYear() {
+        return cy.get('[name="year"]')
+    }
+
+    appointmentHour() {
+        return cy.get('[name="hour"]')
+    }
+
+    appointmentMinute() {
+        return cy.get('[name="minute"]')
+    }
+
+    appointmentAmPmSelector() {
+        return cy.get('[name="amPm"]')
+    }
+
+    documentsSignedButton() {
+        return cy.get('.lbh-button').contains('Documents signed');
+    }
+
+    activityHistoryButton() {
+        return cy.get('.lbh-button').contains('Case activity history');
+    
+    }
+
+    hasNotifiedResident() {
+        return cy.get('[name="hasNotifiedResident"]');
+    }
```

Comment:
Would be good to include what it is within the name as it's unclear ...
Is that a tickbox or a radio, a dropdown, or page title?